trivy-image-scan/action.yml

```diff
@@ -31,46 +27,60 @@ inputs:
     description: 'comma-separated list of relative paths in repository to one or more .trivyignore files'
     required: true
     default: '.trivyignore'
-  format:
-    description: 'output format (table, json, template)'
+  output-mode:
+    description: 'Upload to github security via sarif format (github), or log pretty print output (log)'
     required: true
-    default: 'sarif'
+    default: 'github'
 runs:
   using: "composite"
   steps:
     - uses: gradle/gradle-build-action@v2
     - name: Determine container tag
-      if: "${{ github.event.inputs.tag == '' }}"
+      if: ${{ github.event.inputs.tag == '' }}
       shell: bash
       run: |
         echo "IMAGE_TAG=$(./gradlew -q printDockerImageDefaultTag | head -1)" >> $GITHUB_ENV
 
     - name: Determine container tag
-      if: "${{ github.event.inputs.tag != '' }}"
+      if: ${{ github.event.inputs.tag != '' }}
       shell: bash
       run: |
         echo "IMAGE_TAG=${{ inputs.tag }}" >> $GITHUB_ENV
 
+    - name: Set up output values
+      if: ${{ inputs.output-mode == 'github' }}
+      shell: bash
+      run: |
+        echo "FORMAT=sarif" >> $GITHUB_ENV
+        echo "OUTPUT=''trivy-results.sarif''" >> $GITHUB_ENV
+
+    - name: Set up output values
+      if: ${{ inputs.output-mode == 'log' }}
+      shell: bash
+      run: |
+        echo "FORMAT=table" >> $GITHUB_ENV
+        echo "OUTPUT=''" >> $GITHUB_ENV
+
     - name: Default ignore file if not exist
+      if: ${{ inputs.trivyignores == '.trivyignore' }}  # If a non-default ignore is passed, then we can assume it exists
       shell: bash
-      run: | # If a non-default ignore is passed, then we can assume it exists
+      run: |
         touch .trivyignore
 
     - name: Run Trivy vulnerability scanner
       uses: aquasecurity/trivy-action@master
       with:
         trivyignores: ${{ inputs.trivyignores }}
         image-ref: ${{ inputs.image }}:${{ env.IMAGE_TAG }}
-        format: ${{ inputs.format }}
-        output: ${{ inputs.output }}
+        format: ${{ env.FORMAT }}
+        output: ${{ env.OUTPUT }}
         vuln-type: ${{ inputs.vuln-type }}
         severity: ${{ inputs.severity }}
         ignore-unfixed: ${{ inputs.ignore-unfixed }}
         timeout: ${{ inputs.timeout }}
 
     - name: Upload Trivy scan results to GitHub Security tab
       uses: github/codeql-action/upload-sarif@v2
-      if: always()
-      continue-on-error: true # Not required if security not enabled
+      if: ${{ always() && inputs.output-mode == 'github' }}
```

Comment:
`always() &&` can be removed.

Reply:
So what I'm trying to do here is make this always run, regardless of failure, but only if github output mode. If log output mode, it should never run.

Reply:
Without the always, does just the condition alone cause it to ignore the success of earlier steps?

Reply:
As per github doc, status check function is required in the expression. otherwise, it will add default `success()` which requires all preceding steps to success.
Another recommendation is to use `success() || failure()` instead of `always()` to handle canceled jobs.

https://docs.github.com/en/actions/learn-github-actions/expressions#status-check-functions